artistools/inputmodel/inputmodel_misc.py

```diff
@@ -379,6 +379,7 @@ def add_derived_cols_to_modeldata(
         get_cell_angle(dfmodel, modelpath)
 
     if "propcells" in derived_cols:
+        assert dimensions == 3  # Only works for 3D case. 1D or 2D can have multiple associated cells
```

Comment:
In the 3D case, the propagation cell indices are trivially related to the model grid indicies (equal unless the cell is empty). The grid mapping file exists to make volume corrections for the 1D and 2D models, since the sum of the associated Cartesian block volumes is not an perfect match to the spherical shell or 2D ring volume.

Reply:
In Classic ARTIS they are not trivially related even in 3D.